### PR TITLE
Fixes #14692: rudder agent check does not repair policies if they are broken 

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/14692

### CODE REVIEW

File: share/commands/agent-check [hunk 1]
```diff
@@ -248,7 +248,7 @@ check_and_fix_inputs() {
   fi
 
   # Ensure main promises are not broken, and clean ncf cache and force a new download if so
-  if ${CFE_BIN_DIR}/cf-promises | grep error > /dev/null
+  if ! ${CFE_BIN_DIR}/cf-promises > /dev/null
```
Comment (amousset) [inline share/commands/agent-check]: Are you sure it returns !=0 when it fails

Comment (ncharles) [reply share/commands/agent-check]: yes, on my tests that's the case

Comment (ncharles) [reply share/commands/agent-check]: at least on similares version of rudder

**Comment by ncharles** [comment]
do not merge, it should target 4.3

**Comment by ncharles** [comment]
PR replaced by https://github.com/Normation/rudder-agent/pull/210

**Pull request closed without merging**